src/main/java/org/moditect/jfrunit/JfrEvents.java

```diff
@@ -305,4 +326,19 @@ private String getDumpFileName() {
     private String getDefaultDumpFileName() {
         return testMethod.getDeclaringClass().getName() + "-" + testMethod.getName() + ".jfr";
     }
+
+    public JfrAnalysisResults automaticAnalysis() {
+        if (analysis == null) {
```

Comment:
Is there a reason for this check? I think it would be ok to just let user create multiple analysis if he wants to. I guess it would be also easy to document rather than documenting that he will get the old one if he called the method before.

Reply:
Originally, this was to cache the results and not re-run the analysis.  But as you mention this complicates behaviour.  I will remove this and let the user cache the result if they need to

Reply:
The analysis is no longer cached in `JfrEvents`